### PR TITLE
ed: Symlink ed to ged with --default-names

### PR DESCRIPTION
**Pull request by sjackman**
Symlink `ed` to `ged` when compiled with `--default-names` to fix `test`.

### CODE REVIEW

**Comment by ilovezfs** [comment]
This doesn't seem like the right way to fix the bug. If I'm using default names, there shouldn't be any g+ prefix around at all. Is this the bug where superenv fails to pick the brew version when it has the default name? Or is it a different bug?

**Comment by sjackman** [comment]
`brew test` fails because it executes `ged`, which is named `ed` if you install it `--with-default-names`.

Any scripts that depend on GNU `ed` being named `ged` on Homebrew will fail if it's installed `--with-default-names`. This fix ensures that those scripts will continue to work.

**Comment by sjackman** [comment]
```
Error: 1 problem in 1 formula
homebrew/dupes/ed:
  * Stable: Please use https:// for http://ftpmirror.gnu.org/ed/ed-1.12.tar.lz
```
http://bot.brew.sh/job/Homebrew%20Dupes%20Pull%20Requests/311/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/audit_homebrew_dupes_ed/

**Comment by ilovezfs** [comment]
@sjackman I think I'd change the test to check the build options and then execute the correct binary. For example, https://github.com/Homebrew/homebrew-core/blob/master/Formula/idris.rb#L56-L63

**Comment by sjackman** [comment]
Done. Also fixed the above audit warning.

**Comment by sjackman** [comment]
@ilovezfs Bump. =)

**Comment by ilovezfs** [comment]
@sjackman looks good. You might also want to add a third parameter to the `pipe_output`s asserting the exit status is zero, since `pipe_output` won't do that by default, unlike `shell_output`, which does, though it's hard to imagine the replacements succeeding without the exit status having been zero, but you never know!

**Comment by sjackman** [comment]
Fixed. Thanks for the suggestion, Joe.

**Comment by ilovezfs** [comment]
:+1:

File: ed.rb [hunk 1]
```diff
@@ -37,7 +37,11 @@ def caveats
   test do
     testfile = (testpath/"test")
```
Comment (ilovezfs) [inline ed.rb]: This doesn't actually need the parentheses if you feel like removing them

Comment (sjackman) [inline ed.rb]: Fixed.

**Comment by ilovezfs** [comment]
@DomT4 :ship:?

**Comment by ilovezfs** [comment]
@sjackman I guess to be consistent, this seems to be the idiom:
https://github.com/Homebrew/homebrew-dupes/blob/1542997258b0fca605c95a3df23ea200cbb9bff5/grep.rb#L50
https://github.com/Homebrew/homebrew-dupes/blob/046ed555453d7f464c044d2da0f83764c614feaf/make.rb#L37

**Comment by DomT4** [comment]
Prefer the way ILZ points out above, but in theory 👍.

**Comment by sjackman** [comment]
Agreed. I recall that I have had pushback against using the ternary operator in formulas, but I like it just fine. Fixed.

**Comment by DomT4** [comment]
Mike's not a fan of them, but I think handling default names in tests is a reasonable use case.

**Comment by sjackman** [comment]
If it's for readability, (because `?` and `:` look weird), I quite like using if/then/else/end, though it's more verbose.
```ruby
cmd = build.with?("default-names") ? "ed" : "ged"
cmd = if build.with?("default-names") then "ed" else "ged" end
```

**Comment by DomT4** [comment]
It's more that ternary operators can be obtuse to people new to Ruby, and formulae are _supposed_ to be fairly easy.

**Comment by ilovezfs** [comment]
I think we can expect "them" to learn the ternary operator. If we need to use stick figures and crayons to help, I'm not opposed.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
pulled.

**Comment by sjackman** [comment]
Thanks, Joe.

**Pull request closed without merging**